pkg/resource/deploy/providers/registry.go

```diff
@@ -254,6 +254,15 @@ func (r *Registry) Check(urn resource.URN, olds, news resource.PropertyMap,
 	return inputs, nil, nil
 }
 
+// RegisterAlias updates the registry such that the new URN points to the aliased provider
+func (r *Registry) RegisterAlias(alias, urn resource.URN) {
+	for ref, prov := range r.providers {
+		if ref.urn == alias {
+			r.setProvider(mustNewReference(urn, ref.id), prov)
+		}
+	}
```

Comment:
In principle we shouldn't need to do this. The caller can know whether or not the aliased resource is a provider by checking the type portion of the URN. I'd prefer that we be more precise here and do those checks in the step generator s.t. the contract for this method can be "alias this provider to this URN, and a provider with the alias target must exist". I think that will make the behavior clearer in both cases.